replication-api/src/main/java/com/connexta/replication/queue/TaskInfo.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.replication.queue;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+/** Represents a task that can be queued for later processing by a worker. */
+public interface TaskInfo {
+  /**
+   * Gets the priority for this task.
+   *
+   * @return the priority for the task
+   */
+  public PriorityLevel getPriority();
+
+  /**
+   * Gets the identifier of the intel document this task is for.
+   *
+   * @return the intel document identifier
+   */
+  public String getId();
+
+  /**
+   * Gets the operation to be performed.
+   *
+   * @return the task operation
+   */
+  public OperationType getOperation();
+
+  /**
+   * Gets the last modified timestamp from the source site for the intel document referenced by this
+   * task.
+   *
+   * @return the last modified timestamp for the intel document
+   */
+  public long getLastModified();
+
+  /**
+   * Gets the information for the resource that should be transferred or empty if the resource
+   * should not be transferred.
+   *
+   * @return the info for the resource that should be transferred or empty if none should be
+   *     transferred
+   */
+  public Optional<ResourceInfo> getResource();
+
+  /**
+   * Gets the information for the different types of metadata that should be transferred.
+   *
+   * @return all the different types of metadata that should be transferred or empty if none should
```

Comment:
❓ Can this be empty? Would we ever transfer a resource without corresponding metadata?

Reply:
Maybe. For Ion to Ion (non tactical) we might just transfer the resource and let the other side regenerate the metadata. The current API allows us the freedom to support both when we have a need to make that decision.